infrastructure/zk/src/init.ts

```diff
@@ -138,6 +136,64 @@ export async function submoduleUpdate() {
     await utils.exec('git submodule update');
 }
 
+function updateConfigFile(path: string, modeConstantValues: any) {
+    let content = fs.readFileSync(path, 'utf-8');
+    let lines = content.split('\n');
+    let entries = Object.entries(modeConstantValues);
+    let addedContent;
+    while (entries.length > 0) {
+        const [key, value] = entries.pop()!;
+        let lineIndex = lines.findIndex((line) => !line.startsWith('#') && line.includes(`${key}=`));
+
+        if (lineIndex !== -1) {
+            if (value !== null) {
+                lines.splice(lineIndex, 1, `${key}=${value}`);
+            } else {
+                lines.splice(lineIndex, 1);
+            }
+        } else {
+            if (value !== null) {
+                addedContent = `${key}=${value}\n`;
+            }
+        }
+    }
+    content = lines.join('\n');
+    if (addedContent) {
+        content += addedContent;
+    }
+    fs.writeFileSync(path, content);
+}
+function updateChainConfig(validiumMode: boolean) {
+    const modeConstantValues = {
+        pubdata_overhead_part: validiumMode
+            ? constants.VALIDIUM_PUBDATA_OVERHEAD_PART
+            : constants.ROLLUP_PUBDATA_OVERHEAD_PART,
+        batch_overhead_l1_gas: validiumMode
+            ? constants.VALIDIUM_BATCH_OVERHEAD_L1_GAS
+            : constants.ROLLUP_BATCH_OVERHEAD_L1_GAS,
+        max_pubdata_per_batch: validiumMode
+            ? constants.VALIDIUM_MAX_PUBDATA_PER_BATCH
+            : constants.ROLLUP_MAX_PUBDATA_PER_BATCH
+    };
+    updateConfigFile(CHAIN_CONFIG_PATH, modeConstantValues);
+}
+function updateEthSenderConfig(validiumMode: boolean) {
+    const modeConstantValues = {
+        internal_enforced_l1_gas_price: validiumMode
+            ? constants.VALIDIUM_ENFORCED_L1_GAS_PRICE
+            : constants.ROLLUP_ENFORCED_L1_GAS_PRICE
+    };
+    updateConfigFile(ETH_SENDER_PATH, modeConstantValues);
+}
```

Comment:
Probably missed this on Friday (long day), but I have a comment here, the `internal_enforced_l1_gas_price` is a variable we override with `45_000_000_000` because in Ethereum's mainnet this is the value for the gas price. We set it like that to see better results when comparing Validium mode vs Rollup mode. By this, I mean that we need to get rid of updating the `eth_sender_config` with this (it is ok to have it hardcoded in our PRs but then this will be reverted in the PR to `main`).

Reply:
Also, it is the same for both modes.

Reply:
👍